src/udiskslinuxmanagernvme.c

```diff
@@ -0,0 +1,505 @@
+      goto out;
+    }
+
+  if (!udisks_daemon_util_check_authorization_sync (manager->daemon,
+                                                    NULL,
+                                                    "org.freedesktop.udisks2.nvme-connect",
+                                                    arg_options,
+                                                    /* Translators: Shown in authentication dialog when the user
+                                                     * requests connection to a NVMeoF controller.
+                                                     */
+                                                    N_("Authentication is required to connect to a NVMe over Fabrics controller"),
+                                                    invocation))
+    goto out;
+
+  g_variant_lookup (arg_options, "transport_svcid", "&s", &transport_svcid);
+  g_variant_lookup (arg_options, "host_traddr", "&s", &host_traddr);
+  g_variant_lookup (arg_options, "host_iface", "&s", &host_iface);
+  g_variant_lookup (arg_options, "host_nqn", "^&ay", &host_nqn);
+  g_variant_lookup (arg_options, "host_id", "^&ay", &host_id);
+  extra_args = fabrics_options_to_extra (arg_options);
+
+  if (!bd_nvme_connect (arg_subsysnqn,
+                        arg_transport,
+                        arg_transport_addr,
+                        transport_svcid,
+                        host_traddr,
+                        host_iface,
+                        host_nqn,
+                        host_id,
+                        (const BDExtraArg **) extra_args,
+                        &error))
+    {
+      g_dbus_method_invocation_take_error (invocation, error);
+      goto out;
+    }
+
+  /* Determine the resulting object */
+  wait_data.subsysnqn = arg_subsysnqn;
+  wait_data.transport = arg_transport;
+  wait_data.transport_addr = arg_transport_addr;
+  wait_data.host_nqn = host_nqn;
+  wait_data.host_id = host_id;
+
+  ctrl_object = udisks_daemon_wait_for_object_sync (manager->daemon,
+                                                    wait_for_fabrics_object,
+                                                    &wait_data,
+                                                    NULL,
+                                                    UDISKS_DEFAULT_WAIT_TIMEOUT,
+                                                    &error);
+  if (ctrl_object == NULL)
+    {
+      g_prefix_error (&error, "Error waiting for NVMeoF controller object: ");
+      g_dbus_method_invocation_take_error (invocation, error);
+      goto out;
+    }
+
+  udisks_manager_nvme_complete_connect (object,
+                                        invocation,
+                                        g_dbus_object_get_object_path (G_DBUS_OBJECT (ctrl_object)));
+
+ out:
+  if (ctrl_object != NULL)
+    g_object_unref (ctrl_object);
+  if (extra_args != NULL)
+    {
+      BDExtraArg **extra_p;
+
+      for (extra_p = extra_args; *extra_p; extra_p++)
```

Comment:
It might be worth adding this to libblockdev. Freeing a list of extra arguments sounds like a frequent use case.

Reply:
--> https://github.com/storaged-project/libblockdev/pull/793